### ISSUE TITLE
Enzyme substrate pairing prediction label

### ISSUE THREAD

**Issue by lpkoh** [Hong-yu-Zhang/OmniESI#4]
Hi, this might be a silly question, but in esp dataset, is the Y value 1 means the enzyme and substrate can pair, and 0 means cannot pair?

The reason we ask is because we use the model on our own dataset of enzyme, and we found negative correlation between activity level and pairing prediction value (higher activity enzyme-substrate pairs have lower prediction value), and the result is very significant in p-value. This seems counterintuitive.

**Comment by Hong-yu-Zhang** [comment]
Hi @lpkoh 
1. Yes, 1 stands for pairing and vice versa
2. This sames strange, esp is the largest and most representative datasets and  benchmark according to our knowledge, it should enable the generalization ability to some degree. Could you share the running command ?

**Comment by Hong-yu-Zhang** [comment]
@lpkoh 
Okay, I get it！Here are some of my thoughts.
1. The training of OmniESI dose not involve explicit guidance about "activity degree of enzyme & substrate". In this case, training samples Q1 & Q2 & Q3 contribute equal gradients in the training BCE loss. Hence, the prediction score is not explicit correlated with activity degree, but correlated with "whether can pair". This might explain why the prediction scores of OmniESI are not sensitive to the activity degree, because OmniESI only think Q1-Q3 can pair (scores all above 0.69).
2. For Q4 cases, I think OmniESI's predictions are False Negative, the reason is like you mentioned: "There are only small differences in the sequence". 
3. When using OmniESI pair prediction function in the wet lab experiment, I think it is safe to drop some unpair candidates and discover some potential pair ones. But when it comes to more finegrained property like activity degree, other metric might be needed.

**Comment by Hong-yu-Zhang** [comment]
@lpkoh 
I think this might be the main reason. You can try to finetune OmniESI on your custom data. This might improve the performance. If you have other content to discuss. Feel free to reopen the issue!
